joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/StaticCallLinker.scala

```diff
@@ -1,64 +1,51 @@
 package io.joern.x2cpg.passes.callgraph
 
+import io.joern.x2cpg.utils.LinkingUtil
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.nodes.*
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, EdgeTypes}
-import io.shiftleft.passes.CpgPass
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import io.shiftleft.semanticcpg.language.*
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
+implicit val resolver: ICallResolver = NoResolve
+class StaticCallLinker(cpg: Cpg) extends ForkJoinParallelCpgPass[Seq[Method]](cpg) with LinkingUtil {
```

Comment:
I suppose in my initial comment, I meant `ForkJoinParallelCpgPass[Method]` in an unbatched way

Reply:
But I guess this works too at reducing query count